### DISCUSSION TITLE
How to append files in multiple files field?

### DISCUSSION THREAD

**Discussion by lamaster** [pocketbase/pocketbase#4813] [Q&A]
Docs say I can't https://pocketbase.io/docs/collections/, I can only remove. `files+` does not work. 
But admin ui has append files feature

**Comment by ganigeorgiev** [comment]
For multiple files field you can simply upload the new file(s) to the `yourFileField` key. See also the note in https://pocketbase.io/docs/files-handling/#uploading-files.

> When you upload a new file to a **single file upload field** (aka. Max Files option is 1) PocketBase will automatically delete the previous uploaded file (if any) and upload the new one in its place.
>
> When you upload a new file to a **multiple file upload field** (aka. Max Files option is > 1) the new file will be appended to the existing field values (as long as the Max Files limit is not reached, otherwise an error will be thrown).

The field modifiers will be normalized with the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355).

**Reply by synapse** [reply]
@ganigeorgiev I am doing something like the code below, but it is always overwriting the already saved files and keeping only the new ones. Am I missing something?
I'm using the latest version 0.25.8

``` js
const form = new FormData();
if (Array.isArray(paths)) {
    for (const file of paths) {
        const fileStream = fs.createReadStream(file);
        form.append(name, fileStream, path.basename(file));
    }
} else {
    const fileStream = fs.createReadStream(paths);
    form.append(name, fileStream, path.basename(paths));
}


instance.patch(`${account.server}/api/collections/${collection}/records/${id}`, form, {
    uploadHeaders: form.getHeaders(),
    maxContentLength: Infinity,
    maxBodyLength: Infinity,
});
```

**Reply by ganigeorgiev** [reply]
>  it is always overwriting the already saved files and keeping only the new ones.

@synapse Yes, this is an intended behavior introduced with the refactoring in PocketBase v0.23+.

If you want to preserve the existing files, then use the `+` suffix/prefix modifier for prepend/append type of operation, aka. replace `name` wit `name + "+"`.

**Reply by synapse** [reply]
Worked perfectly, thanks @ganigeorgiev. Sorry for missing the new refactor changelogs

**Reply by synapse** [reply]
Hey @ganigeorgiev sorry for getting back on this, just wanted to know if there's a way to know after a files is append which one is it. Let me rephrase the question: when I append a file I get back the whole records data including the list of all the files (which are renamed). Is there a way to know which file was just uploaded?
Second question: is there a way from the API to grab the file's info (like size, mime, etc)?

Thanks in advance

**Reply by synapse** [reply]
@ganigeorgiev can you help me out with this one 🙏 ☝️ ?
